src/services/refactors/moveToNewFile.ts

```diff
@@ -351,6 +351,8 @@ namespace ts.refactor {
             if (namedBindings) {
                 if (namedBindingsUnused) {
                     changes.delete(sourceFile, namedBindings);
+                    // Also include the ", ":
+                    changes.replaceRangeWithText(sourceFile, { pos: namedBindings.pos - 1, end: namedBindings.pos }, "");

```

Comment:
I would probably opt for

```ts
changes.replaceNode(
  sourceFile,
  importDecl.importClause,
  updateImportClause(importDecl.importClause, name, /*namedBindings*/ undefined));
```

(to replace both lines in this `if`)

Reply:
I think the issue is that the comma token as a child of the ImportClause, and TextChanges only does the exact thing you tell it to do without being smart about the context it’s working in. As you discovered, deleting the named bindings node just removes the text from `pos` to `end`, which doesn’t include the comma. So really, it's the ImportClause that needs to change—it needs to drop its named bindings node _and_ its comma token. Replacing it with a new one will re-print its contents, which lets the emitter be smart about whether a comma is needed without requiring manual intervention.

Reply:
Perfect, yep - moving up the AST makes a lot of sense. 

That change makes a lot of sense. Thanks 👯‍♂